File: waypoints/src/main/java/com/skedgo/android/tripkit/waypoints/WaypointSegmentAdapter.java
package com.skedgo.android.tripkit.waypoints;

import android.support.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import java.util.List;

@Gson.TypeAdapters
@Value.Immutable
@JsonAdapter(GsonAdaptersWaypointSegmentAdapter.class)
public abstract class WaypointSegmentAdapter {

  public abstract String start();
  public abstract String end();
  public abstract List<String> modes();

  @Value.Default public int startTime() {return 0;}

  @Value.Default public int endTime() {return 0;}

  @Nullable public abstract String serviceTripId();
  @Nullable public abstract String operator();
  @Nullable public abstract String region();

}

File: waypoints/src/main/java/com/skedgo/android/tripkit/waypoints/WaypointSegmentAdapter.kt
package com.skedgo.android.tripkit.waypoints

class WaypointSegmentAdapter(
        val start: String,
        val end: String,
        val modes: List<String>,
        val startTime: Int = 0,
        val endTime: Int = 0,
        val serviceTripId: String = "",
        val operator: String = "",
        val region: String = ""
)